packages/core/mesh/network-manager/src/swarm/peer.ts

```diff
@@ -189,21 +193,51 @@ export class Peer {
         log('ignoring response');
         return;
       }
+    } catch (err: any) {
+      log('initiation error: send offer', { err, topic: this.topic, peerId: this.localPeerId, remoteId: this.id });
+      await connection.abort(err);
+      throw err;
+    } finally {
+      this.initiating = false;
+    }
 
+    try {
       if (!answer.accept) {
         this._callbacks.onRejected();
         return;
       }
+    } catch (err: any) {
+      log('initiation error: accept answer', {
+        err,
+        topic: this.topic,
+        peerId: this.localPeerId,
+        remoteId: this.id,
+      });
+      await connection.abort(err);
+      throw err;
+    } finally {
+      this.initiating = false;
+    }
+
+    try {
+      log('opening connection as initiator');
       await connection.openConnection();
       this._callbacks.onAccepted();
     } catch (err: any) {
-      log('initiation error', { err, topic: this.topic, peerId: this.localPeerId, remoteId: this.id });
+      log('initiation error: open connection', {
+        err,
+        topic: this.topic,
+        peerId: this.localPeerId,
+        remoteId: this.id,
+      });
       if (err instanceof TimeoutError) {
         log.warn('aborting connection due to signalling failure');
```

Comment:
I think there's still a chance for signalling failure since the transport isn't fully connected yet. but will review/refresh the error message/handling